.github/workflows/ci.yml

```diff
@@ -118,17 +175,73 @@ jobs:
         python-version: ${{ matrix.python-version }}
         allow-prereleases: true
 
+    - name: Set up Deno
+      uses: denoland/setup-deno@v2
+
     - name: Setup uv
       uses: astral-sh/setup-uv@v3
       with:
         enable-cache: true
         cache-dependency-glob: pyproject.toml
-        cache-suffix: py${{ matrix.python-version }}
+        cache-suffix: "py${{ matrix.python-version }}-${{ matrix.resolution }}"
 
     - name: Install dependencies
       env:
         UV_RESOLUTION: ${{ matrix.resolution }}
       run: make setup
 
+    - name: Install Deno Jupyter Kernel
+      run: deno jupyter --install
+
+    - name: Fix deno path on macOS
+      if: runner.os == 'macOS'
+      run: |
+        source .venv/bin/activate
+        export JUPYTER_PLATFORM_DIRS=0
+
+        # ensure jupyter can find the kernel config using the old path structure
```

Comment:
style: Please capitalize and punctuate (period at the end) every comment in the PR :pray: